tests/rules/no-missing-space-atx.test.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Tests for no-missing-space-atx rule.
+ * @author Sweta Tanwar (@SwetaTanwar)
+ */
+
+//------------------------------------------------------------------------------
+// Imports
+//------------------------------------------------------------------------------
+
+import rule from "../../src/rules/no-missing-space-atx.js";
+import markdown from "../../src/index.js";
+import { RuleTester } from "eslint";
+import dedent from "dedent";
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+	plugins: {
+		markdown,
+	},
+	language: "markdown/commonmark",
+});
+
+//------------------------------------------------------------------------------
+// Valid Test Cases
+//------------------------------------------------------------------------------
+
+const validHeadings = [
+	// 1. ATX headings with proper spacing
+	"# Heading 1",
+	"## Heading 2",
+	"### Heading 3",
+	"#### Heading 4",
+	"##### Heading 5",
+	"###### Heading 6",
```

Comment:
html headings should also be valid.
```suggestion
	"###### Heading 6",
	"<h1>Heading 1<h1>",
	"<h2>Heading 2<h2>",
	"<h3>Heading 3<h3>",
	"<h4>Heading 4<h4>",
	"<h5>Heading 5<h5>",
	"<h6>Heading 6<h6>",
	"<h1>#valid heading</h1>"
```

Reply:
Quotes should also be valid:
```suggestion
	"###### Heading 6",
	`'#something'`
	`"#something"`
```

Reply:
> Maybe we can just traverse the paragrph node and check for the first child

@snitin315 This was a great approach—thanks for this suggestion! It really simplified the code and helped avoid unnecessary node traversals. 🙌